src/MongoDB.Driver.Core/Core/Compression/ICompressor.cs

```diff
@@ -23,6 +23,7 @@ namespace MongoDB.Driver.Core.Compression
     public enum CompressorType
     {
         // NOTE: the numeric values of the enum members MUST be kept in sync with the binary wire protocol
+        // NOTE: the compression keys of the enum members MUST be kept in sync with the compressor connection string options
```

Comment:
The simplest solution is to follow a similar way as we already used for mapping between this enum and values expected by wire protocol.

Reply:
As far as I see, Robert's concern is related to the fact that we cannot safely rely that noone from us will try to change/refactor this naming (even though we highlighted it in the comment) and then this naming will be out of sync with what values are expected by the server.

Reply:
If we don't accept this change, we will need to create a mapping between `Compressor.Type` values and the values which we send to the server and use it everywhere in the driver. 
Now we have two places where we will need to do it:
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Connections/IsMasterHelper.cs#L40
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1446